cmd/ipfs/daemon.go

```diff
@@ -108,6 +111,13 @@ func daemonFunc(req cmds.Request) (interface{}, error) {
 		return nil, err
 	}
 
+	// ignore error for gateway address
+	// if there is an error (invalid address), then don't run the gateway
+	gatewayMaddr, _ := ma.NewMultiaddr(cfg.Addresses.Gateway)
+	if gatewayMaddr == nil {
+		fmt.Println("Invalid gateway address, not running gateway")
```

Comment:
might want to just not display anything to the user, since it's normal operation. maybe just `log.Info`? and :+1: on the comment re: error